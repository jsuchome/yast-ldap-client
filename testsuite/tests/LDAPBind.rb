# encoding: utf-8

#  LDAPBind.ycp
#  Test of Ldap:LDAPBind function
#  Author:	Jiri Suchomel <jsuchome@suse.cz>
#  $Id$
module Yast
  class LDAPBindClient < Client
    def main
      Yast.include self, "testsuite.rb"
      # testedfiles: Ldap.ycp

      @READ = {
        "target" => { "size" => -1 },
        "ldap"   => {
          "error"   => { "msg" => "Bind failed" },
          "product" => { "features" => { "EVMS_CONFIG" => "nazdar" } }
        }
      }
      @EX = { "ldap" => true }
      @EX_F = { "ldap" => false }

      TESTSUITE_INIT([@READ, {}, {}], nil)

      Yast.import "Ldap"

      DUMP("==== bind anonymously ==============================")

      Ldap.anonymous = true

      TEST(lambda { Ldap.LDAPBind("pw") }, [{}, {}, @EX], 0)

      Ldap.anonymous = false
      Ldap.bind_dn = "uid=manager,dc=suse,dc=cz"

      DUMP("==== bind failed ====================================")

      TEST(lambda { Ldap.LDAPBind("p") }, [@READ, {}, @EX_F], 0)

      DUMP("==== bind ok ========================================")

      TEST(lambda { Ldap.LDAPBind("pw") }, [{}, {}, @EX], 0)

      nil
    end
  end
end

Yast::LDAPBindClient.new.main
