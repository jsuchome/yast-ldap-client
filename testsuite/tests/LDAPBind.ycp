/**
 *  LDAPBind.ycp
 *  Test of Ldap:LDAPBind function
 *  Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{

    include "testsuite.ycp";
    // testedfiles: Ldap.ycp

    map READ =	$[ "ldap": $[
	"error": $[
	    "msg": "Bind failed",
	],
	"product" : $[ "features" :
	    $[ "EVMS_CONFIG" : "nazdar" ],
	],
    ]];
    map EX =	$[ "ldap": true  ];
    map EX_F =	$[ "ldap": false ];

    TESTSUITE_INIT([READ,$[],$[]], nil);

    import "Ldap";

    DUMP ("==== bind anonymously ==============================");

    Ldap::anonymous = true;

    TEST(``(Ldap::LDAPBind ("pw")), [ $[], $[], EX ], 0);

    Ldap::anonymous = false;
    Ldap::bind_dn = "uid=manager,dc=suse,dc=cz";

    DUMP ("==== bind failed ====================================");

    TEST(``(Ldap::LDAPBind ("p")), [ READ, $[], EX_F ], 0);

    DUMP ("==== bind ok ========================================");

    TEST(``(Ldap::LDAPBind ("pw")), [ $[], $[], EX ], 0);

}
