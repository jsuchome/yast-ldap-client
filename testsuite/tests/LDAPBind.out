Dump	==== bind anonymously ==============================
Execute	.ldap.bind $[] true
Return	
Dump	==== bind failed ====================================
Execute	.ldap.bind $["bind_dn":"uid=manager,dc=suse,dc=cz", "bind_pw":"p"] false
Read	.ldap.error $["msg":"Bind failed"]
Return	Bind failed
Dump	==== bind ok ========================================
Execute	.ldap.bind $["bind_dn":"uid=manager,dc=suse,dc=cz", "bind_pw":"pw"] true
Return	
